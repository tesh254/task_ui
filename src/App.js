import React from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";
import { Provider } from "react-redux";
import jwt_decode from "jwt-decode";
import { ToastContainer } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";
import "./bulma.css";

import setAuthToken from "./helpers/setAuthToken";
import store from "./redux/store.js";
import {
  setCurrentUser,
  logoutUser
} from "./redux/actions/auth/setCurrentUser";
import PrivateRoute from "./HOC/PrivateRouter";
import tokenChecker from "./helpers/tokenChecker";

// Components
import Nav from "./components/commons/Nav";
import LoginView from "./views/LoginView";
import TaskView from "./views/TaskView";
import Default from "./Default"

if (localStorage.getItem("jwt_token")) {
  try {
    // Set auth token header auth
    setAuthToken(localStorage.getItem("jwt_token"));
    // Decode token and get its content
    const decoded = jwt_decode(localStorage.getItem("jwt_token"));
    // Set user and isAuthenticate
    store.dispatch(setCurrentUser(true));
    // Check for expired token
    const currentTime = Date.now() / 1000;

    if (decoded.exp < currentTime) {
      store.dispatch(logoutUser());

      // Redirect to login
      window.location.href = "/login";
    }
  } catch (e) {
    console.log(e);
  }
}

class App extends React.Component {
  componentDidMount() {
    // Check token
    tokenChecker();
  }

  render() {
    return (
      <Provider store={store}>
        <BrowserRouter>
          <ToastContainer autoClose={3000} />
          <div className="App">
            <Nav />
            <Switch>
              <Route exact path="/login" component={LoginView} />
              <PrivateRoute exact path="/tasks" component={TaskView} />
              <Route exact path="/" component={Default} />
            </Switch>
          </div>
        </BrowserRouter>
      </Provider>
    );
  }
}

export default App;
