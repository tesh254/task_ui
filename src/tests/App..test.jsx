import React from "react"
import enzyme, { mount } from "enzyme"
import toJSON from "enzyme-to-json"
import adapter from "enzyme-adapter-react-16"
import App from "../App";

enzyme.configure({ adapter: new adapter() })

describe("test app component", () => {
    it("should render properly", () => {
        const wrapper = mount(
            <App />
        )
        expect(toJSON(wrapper)).toMatchSnapshot()
    })
})