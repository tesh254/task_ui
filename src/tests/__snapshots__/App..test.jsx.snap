// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test app component should render properly 1`] = `
<App>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <BrowserRouter>
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <ToastContainer
          autoClose={3000}
          bodyClassName={null}
          className={null}
          closeButton={
            <CloseButton
              ariaLabel="close"
            />
          }
          closeOnClick={true}
          draggable={true}
          draggablePercent={80}
          hideProgressBar={false}
          newestOnTop={false}
          pauseOnFocusLoss={true}
          pauseOnHover={true}
          position="top-right"
          progressClassName={null}
          progressStyle={null}
          role="alert"
          rtl={false}
          style={null}
          toastClassName={null}
          transition={[Function]}
        >
          <div
            className="Toastify"
          />
        </ToastContainer>
        <div
          className="App"
        >
          <Nav>
            <nav
              aria-label="main navigation"
              className="navbar"
              role="navigation"
            >
              <div
                className="navbar-brand"
              >
                <a
                  className="navbar-item"
                  href="https://bulma.io"
                  style={
                    Object {
                      "fontSize": "18pt",
                      "fontWeight": "bold",
                    }
                  }
                >
                  Tasks
                </a>
                <a
                  aria-expanded="false"
                  aria-label="menu"
                  className="navbar-burger"
                  role="button"
                >
                  <span
                    aria-hidden="true"
                  />
                  <span
                    aria-hidden="true"
                  />
                  <span
                    aria-hidden="true"
                  />
                </a>
              </div>
            </nav>
          </Nav>
          <Switch />
        </div>
      </Router>
    </BrowserRouter>
  </Provider>
</App>
`;
